### DISCUSSION TITLE
confusing voice navigation

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#30] [Q&A] [answered]
For riders who only navigate via voice this could be confusing.

I guess you can take any motorway access and a route example is not necessary.

if a motorway access is like in this picture, the voice instruction is to "keep **left**", but this is wrong in this case because the access is to the right. If you keep left then you miss the motorway access.

![Screenshot_20230203-110915_Cruiser](https://user-images.githubusercontent.com/121371629/216573643-588037ac-76d8-40f4-93cd-e60e5d3aaa99.jpg)

**Comment by devemux86** [comment]
Important: _Turn instructions are generated by the routing services._
Navigation just announces the already calculated turn instructions.

Route examples are useful + the routing service name.

We can check the turn instructions before navigation
in turn instructions list and with yellow marks on map.

Also be careful not to put waypoints near intersections,
or calculated turn instructions may have such effects.

**Comment by WolfgangZ42** [comment]
With the best will, I can't see a problem here - see simulation: [https://youtu.be/vF7AoZivrqo](https://youtu.be/vF7AoZivrqo)

Route: 
[Saarlouis.zip](https://github.com/devemux86/cruiser/files/10578797/Saarlouis.zip)

**Comment by devemux86** [comment]
Thanks for the sample.

If it is a kurviger route,
then the turn instructions are generated by Kurviger / GraphHopper.

They can also be checked on their websites.

**Reply by WolfgangZ42** [reply]
But it's quite the same with a route planned in the Cruiser app: 
[20230203_141030.zip](https://github.com/devemux86/cruiser/files/10579160/20230203_141030.zip)

**Reply by devemux86** [reply]
With what routing service?

It's not unexpected that many routing services generate "similar" turn instructions.

**Reply by WolfgangZ42** [reply]
OpenRouteService

**Comment by DolomitesTom** [comment]
[Cruiser.zip](https://github.com/devemux86/cruiser/files/10580278/Cruiser.zip)
Brouter offline
The voice navigation is two times "keep left" and only in the very last moment to the right.

**Reply by devemux86** [reply]
The sequence of the turn instructions at the ring should be:
- Keep left
- Turn right
- Keep left

Do you mean that?

![Cruiser](https://user-images.githubusercontent.com/3484020/216643541-784ff378-ad73-4f7f-8ae3-c369ff06cb34.png)

**Reply by DolomitesTom** [reply]
I only mean the audio commands. I don't know this menu. Is says 2 times in 430 meter keep left and then in 180m keep left and then in the very last moment turn right.

**Reply by DolomitesTom** [reply]
I would expect it to say in 430 Meter turn right, then again in 180 meter turn right and then again when reaching the turn off "turn right".

For me it is not a problem because I don't navigate without display, but others do.

**Reply by devemux86** [reply]
(the image is from [Desktop Cruiser](https://github.com/devemux86/cruiser/releases), it is easier to get screenshots)

> I would expect it to say in 430 Meter turn right, then again in 180 meter turn right and then again when reaching the turn off "turn right".

There are several turn instructions nearby and they must be announced in order.

Running your sample locally with:
- Settings | Routing | Options | Travel mode: Motorcycle
- Settings | Routing | Default speed | Motorcycle:  45 km/h
- Settings | Voice guidance | Other | Navigation instructions: All checked
- Settings | Voice guidance | Other | Announcement time: 100%

The voice guidance speaks:

- (**before ring**)
- In 280 m, Keep left. Then in 180 m, Turn right
- Keep left
- Turn right
- In 280 m, Keep left
- Keep left
- (**after ring**)
- In 280 m, Keep left

What settings do you use?

**Comment by WolfgangZ42** [comment]
The whole thing is an overlap of the voice announcements when driving with "2x" in the simulation - with "1x" everything is okay.

**Reply by devemux86** [reply]
This can happen when the "Settings | Voice guidance | Other | Two commands at once" is enabled
and there are several turn instructions nearby.

Voice announcements stop the previous one, so there should not be any overlap(?).

**Comment by DolomitesTom** [comment, marked as answer]
Settings | Routing | Default speed | Motorcycle: 80 km/h

Guessing that this is the average speed, isn't it? It is meant to later calculate the tour duration, right? By the way, to my opinion I would name it default *average* speed. My default permanent speed is around 130 km/h (plus minus 20 km/h), of course only outside of cities. In the avarage it is mostly something over 80 km/h.

Other settings same

It was on a real trip, no simulation. I only tried this to simulate yesterday evening. I just find a theoretical problem is this "keep *left*" when the direction is to turn right. This happens pretty often and I thought "why does it now says keep left when I drive straight forward and the next turn is right. 

Maybe it is not worth discussing it. I anyway have the display in the peripheral area of my visual field. I personally can get used to it. I only thought this could be a problem for some users. If not just close this topic.

**Reply by devemux86** [reply]
> It is meant to later calculate the tour duration, right?

Correctly.

Note: the selected speed also affects the voice guidance (see https://github.com/devemux86/cruiser/discussions/19).
Larger speed causes the voice instructions to be announced earlier.

> By the way, to my opinion I would name it default average speed.

Good idea, seems a more appropriate name.

Regarding the instructions, the issue is that the app / navigator receives them from the routers.
Some routers generate "many" instructions. We could skip some, but they may be useful elsewhere.

**Reply by DolomitesTom** [reply]
Difficult if different routes create different instructions. Thanks für explanation. I guess it's ok then, no need to adapt something.
